it/__init__.py

```diff
@@ -0,0 +1,187 @@
+
+
+class ConfigFile:
+    def __init__(self, config_name):
+        self.user_home = os.path.expanduser("~")
+        self.rally_home = os.path.join(self.user_home, ".rally")
+        self.config_file_name = "rally-{}.ini".format(config_name)
+        self.source_path = os.path.join(os.path.dirname(__file__), "resources", self.config_file_name)
+        self.target_path = os.path.join(self.rally_home, self.config_file_name)
+
+
+class TestCluster:
+    def __init__(self, cfg):
+        self.cfg = cfg
+        self.installation_id = None
+        self.http_port = None
+
+    def install(self, distribution_version, node_name, http_port):
+        self.http_port = http_port
+        transport_port = http_port + 100
+        try:
+            output = process.run_subprocess_with_output(
+                "esrally install --configuration-name={cfg} --quiet --distribution-version={dist} --build-type=tar "
+                "--http-port={http_port} --node={node_name} --master-nodes={node_name} "
+                "--seed-hosts=\"127.0.0.1:{transport_port}\"".format(cfg=self.cfg,
+                                                                     dist=distribution_version,
+                                                                     http_port=http_port,
+                                                                     node_name=node_name,
+                                                                     transport_port=transport_port))
+
+            self.installation_id = json.loads("".join(output))["installation-id"]
+        except BaseException as e:
+            raise AssertionError("Failed to install Elasticsearch {}.".format(distribution_version), e)
+
+    def start(self, race_id):
+        cmd = "start --runtime-jdk=\"bundled\" --installation-id={} --race-id={}".format(self.installation_id, race_id)
+        if esrally(self.cfg, cmd) != 0:
+            raise AssertionError("Failed to start Elasticsearch test cluster.")
+        es = client.EsClientFactory(hosts=[{"host": "127.0.0.1", "port": self.http_port}], client_options={}).create()
+        client.wait_for_rest_layer(es)
+
+    def stop(self):
+        if self.installation_id:
+            if esrally(self.cfg, "stop --installation-id={}".format(self.installation_id)) != 0:
+                raise AssertionError("Failed to stop Elasticsearch test cluster.")
+
+
+class EsMetricsStore:
+    VERSION = "7.6.0"
+
+    def __init__(self):
+        self.cluster = TestCluster("in-memory-it")
+
+    def start(self):
+        self.cluster.install(distribution_version=EsMetricsStore.VERSION, node_name="metrics-store", http_port=10200)
+        self.cluster.start(race_id="metrics-store")
+
+    def stop(self):
+        self.cluster.stop()
+
+
+def install_integration_test_config():
+    def copy_config(name):
+        f = ConfigFile(name)
+        io.ensure_dir(f.rally_home)
+        with open(f.target_path, "w", encoding="UTF-8") as target:
```

Comment:
Would it make sense to just read src into `contents` with something:

```
with open(f.source_path, "r", encoding="UTF-8") as src:
    contents = src.read()
```

and then replace using `Template`'s native capabilities, to avoid pure string based replacements, i.e.

```
from string import Template
with open(f.target_path, "w", encoding="UTF-8") as target:
    target.write(Template(contents).substitute(USER_HOME=f.user_name)
```

[AFAIK](https://docs.python.org/3/library/string.html#template-strings) `${identifier}`-like template strings are supported out of the box.

Reply:
Thanks! Much better. Addressed in https://github.com/elastic/rally/pull/919/commits/eb03450a0fa37fe5ad06ea2225736d6fedd8e32d.